components/dashboard/src/App.tsx

```diff
@@ -40,6 +42,33 @@ function App() {
                 setUser(await getGitpodService().server.getLoggedInUser());
             } catch (error) {
                 console.log(error);
+                if (error instanceof ResponseError) {
+                    switch (error.code) {
+                        case ErrorCodes.USER_BLOCKED:
+                            window.location.href = gitpodHostUrl.with({ pathname: "/blocked" }).toString();
```

Comment:
These should all probably just be components / routes. Since we have a single entry-point, redirecting here will just infinite loop.